modules/preview-react/segmented-control/stories/SegmentedControl.stories.mdx

```diff
@@ -38,62 +37,86 @@ yarn add @workday/canvas-kit-preview-react
+components have an `icon` prop and/or text content: icon-only, text-only, and text-and-icon.
 
-<ExampleCodeBlock code={TextAndIcon} />
+All `SegmentedControl.Item` components within a given `SegmentedControl` must be of the same
+variation.
+
+#### Icon-Only
 
-#### Text Only
+To render an icon-only `SegmentedControl`, apply the `icon` prop to `SegmentedControl.Item` and do
+not provide it with text content. Refer to the [basic example](#basic-example) above for an instance
+of an icon-only `SegmentedControl`.
+
+The icon-only variation is the only variation which supports a vertical orientation in addition to
+the default horizontal orientation. Set the `orientation` prop of `SegmentedControl` to `vertical`
+to configure the component to render vertically.
+
+<ExampleCodeBlock code={Vertical} />
+
+#### Text-Only
+
+To render a text-only `SegmentedControl`, omit the `icon` prop from `SegmentedControl.Item` and
+provide it with text content.
 
 <ExampleCodeBlock code={TextOnly} />
 
-#### Icon Only (vertical)
+#### Text-and-Icon
 
-`SegmentedControl` icon only buttons can be placed horizontally or vertically. Be aware that
-vertical orientation must be used only for icon only buttons.
+To render a text-and-icon `SegmentedControl`, apply the `icon` prop to `SegmentedControl.Item` and
+provide it with text content.
 
-<ExampleCodeBlock code={Vertical} />
+<ExampleCodeBlock code={TextAndIcon} />
+
+### Sizes
 
-### Sizes Example
+`SegmentedControl` accepts a `size` prop which supports the following values:
 
-`SegmentedControl` has three different size: `small`, `medium` (default size) and `large`.
+- `small`
+- `medium` (Default)
+- `large`
 
 <ExampleCodeBlock code={Sizes} />
 
-### Disabled Example
+### Disabled
 
-`SegmentedControl`can be disabled by adding`disabled` prop to the whole container. This will disable
-all buttons.
+Set the `disabled` prop of `SegmentedControl` to disable the entire component including its buttons.
 
 <ExampleCodeBlock code={Disabled} />
 
-### RTL Example
+### Right-to-Left (RTL)
 
 `SegmentedControl` supports right-to-left languages when specified in the `CanvasProvider` `theme`.
 
 <ExampleCodeBlock code={RTL} />
 
-### Dynamic Items Example
+### Dynamic Items
 
-The `SegmentedControl.Item` component takes in an optional `index` property if you want to specify
-the position of a button. If not defined, by default it will append items to the end. In this
-example, our items are stored as an array in the state.
+`SegmentedControl` supports a
+[dynamic API](/getting-started/for-developers/resources/collection-api/#dynamic-items) where instead
+of statically providing the JSX for each `SegmentedControl.Item`, you pass an array of `items` in
+the `model` state and provide a render function to display the items.
 
 <ExampleCodeBlock code={Dynamic} />
 
+Note that `SegmentedControl.Item` accepts an optional `index` prop if you want to specify the
+position of the item. Items are appended to the end by default.
```

Comment:
@RayRedGoose I was confused a bit by this `index` prop on `SegmentedControl.Item` (I noticed you lifted this language from the Tabs docs, perhaps we need to revisit those docs as well). I couldn't get it to work with the static API or the dynamic API -- could you provide a code snippet or example illustrating how to use it?